src/sp/it/pl/util/file/Util.kt

```diff
@@ -130,6 +130,30 @@ fun File.toURLOrNull() =
             null
         }
 
+/**
+ *  Invokes the specified block that saves the file with safe temporary file saving semantics:
+ *  - 1 remove temporary file if exists
+ *  - 2 rename existing file to temporary file
+ *  - 3 invoke block to save the file
+ *  - if 3 succeeds remove the temporary file
+ *  - if 3 fails rename the temporary file to original file
+ */
+fun <OK,ERROR> File.writeSafely(block: (File) -> Try<OK,ERROR>): Try<OK,ERROR> {
```

Comment:
I'm not sure if this can not cause problems, i.e., deleting the original file should be the last operation, otherwise there is a risk of losing the data. Renaming a file is probably same, in that the worst case is user would need to rename the file correctly to get back his data, but I'm not sure. Is renaming file 100% safe in regards to the data of the file? If not then I would avoid your solution.

Reply:
Yes, renaming a file only touches the pointer, not the data. Since we only delete the original file after writing the tmp file, the risk of data loss is even less, because we already have a valid new file before touching the original at all.

Reply:
Alright. I'm still concerned however. What the save operation does is produce new data. Having the first operation be renaming the old data to tmp makes it clear to every potential future reader that saving took place and in worse cases scenario, there is tmp data to rely on.

How does your model behavein this regard? We would write to tmp file and if that failed, we could end up with the original data still there and corrupted data in a temp. Not good. Although what is good is that for potential future reader this safe writing is fully transparent regardless of the outcome. So we would need to only add some error handling, like deleting tmp if saving fails.

This brings up another issue: user should be notified of save failure. He will probably not be able to do anything, but at least he will know. However notifying during application closing can put us in corrupted state, as some services/components may be disposed of.

Reply:
> We would write to tmp file and if that failed, we could end up with the original data still there and corrupted data in a temp. Not good. 

Why would that be not good? At least we still have the original data to rely on. What I want is to have valid data for as long as possible. 
With my approach, there is only a timespan of nanoseconds where there is no valid data, whereas with your approach the file is invalid for the whole timespan of the writing process.

Reply:
I tried to implement this, but this approach has serious issues. More on that in the upcoming PR.